### ISSUE TITLE
[ Navbar Dropdown Menu ] focus bug when hovering over dropdown items

### ISSUE THREAD

**Issue by JayDijkstra** [materializecss/materialize#72]
## Context

It seems that there is an issue which shows an ugly black/white border when hovering over and then out of a dropdown element which is defined within a navigation bar. It seems that this has something to do with focusing in and out of the dropdown trigger. 

I have attached a screenshot which shows the border:

![image](https://user-images.githubusercontent.com/6065344/109795007-be5fb480-7c16-11eb-8ba1-3e2ccf4a6eb7.png)

## Steps to Reproduce (for bugs)

1. Using the following code (as documented on https://materializecss.com/navbar.html#navbar-dropdown) : 

````html
<!-- Dropdown Structure -->
<ul id="dropdown1" class="dropdown-content">
  <li><a href="#!">one</a></li>
  <li><a href="#!">two</a></li>
  <li class="divider"></li>
  <li><a href="#!">three</a></li>
</ul>
<nav>
  <div class="nav-wrapper">
    <a href="#!" class="brand-logo">Logo</a>
    <ul class="right hide-on-med-and-down">
      <li><a href="sass.html">Sass</a></li>
      <li><a href="badges.html">Components</a></li>
      <!-- Dropdown Trigger -->
      <li><a class="dropdown-trigger" href="#!" data-target="dropdown1">Dropdown<i class="material-icons right">arrow_drop_down</i></a></li>
    </ul>
  </div>
</nav>

````
2. Using the following options for the DropDown initialisation:

````js
        document.addEventListener('DOMContentLoaded', function() {
            let elems = document.querySelectorAll('.dropdown-trigger');
            let options = {
                hover: true,
                coverTrigger: false,
            }
            let instances = M.Dropdown.init(elems, options);
        })
````

3. Hover over and out of the dropdown element.

4. Within the developer-console, adding the `:focus` state to the `<li>`, which surrounds the Dropdown Trigger, will result to the same behaviour. 

## Your Environment

* Version used: This issue occurred to me in both `"@materializecss/materialize": "^1.0.0"` and the CDN stated in the documentation (https://materializecss.com/getting-started.html).
* Browser Name and version: Google Chrome Version 88.0.4324.182 (Official Build) (64-bit)
* Operating System and version (desktop or mobile): Linux Mint 20

**Comment by Smankusors** [comment]
is this also reproducible on other browsers such as Firefox?

**Comment by JayDijkstra** [comment]
@Smankusors in Firefox the border is less visible (white-dashed), but does also occur. It can be reproduced by adding the `:focus-visible` state, to the `<li>` parent of the dropdown-trigger, in the developer console.

**Comment by JayDijkstra** [comment]
What I did forget to mention is that a similar (same) issue has been created in https://github.com/Dogfalo/materialize/issues/6051 back in 2018. This comment is just for giving you as much information as I can.

**Comment by Smankusors** [comment]
I think this come from the browser styling. Perhaps this can be solved by simply removing the outline (which is bad for accessibility), or create own outline style when the element focused.

Temporary, you can add CSS rule like `outline: none` to remove the built in outline.

**Comment by JayDijkstra** [comment]
@Smankusors thanks for your solution. The following override works as a (temporarily) fix in both Google Chrome and Mozilla Firefox:

````css

li a.dropdown-trigger:focus {
    outline: none;
}

````

**Comment by wuda-io** [comment]
Hey, nice that you found a solution.
Can you put this in a PR?
Thanks

**Comment by JayDijkstra** [comment]
Dear @wuda-io, by forking the project and following the contribution guidelines to create a Pull-Request, I get a lot of merge-conflicts on my branch when trying to pull in the upstream `master` branch. The branch itself is created of the `v1-dev` because other pull-requests are based on this same branch. 

It is not clear to me if my branch needs to  be created of the `master` or `v1-dev` branch.

**Comment by wuda-io** [comment]
Yes you are right! The correct approach is, to use is the default branch **v1-dev**, and NOT the master. The documentation needs to be updated.
Thanks for your hint.

**Comment by wuda-io** [comment]
Sorry, I thought this was a bug. But as @Smankusors and @DanielRuf suggested it is intended behavoir when using keyboard only. Nevertheless, thank you for your contribution in first place. At least we found out that we have to update the docs. Cheers mate.

**Comment by Smankusors** [comment]
> Sorry, I thought this was a bug. But as @Smankusors and @DanielRuf suggested it is intended behavoir when using keyboard only. Nevertheless, thank you for your contribution in first place. At least we found out that we have to update the docs. Cheers mate.

yeah the styling is intended behavior. But when this is applied without using any keyboard is kinda not expected. Unfortunately I can't reproduce this like @JayDijkstra's screenshot without adding `:focus` state on Chrome 89.

But on Firefox 87.0b9, I got 1px dashed rectangle like he described.

Also I haven't tried to reproduce this on Linux system yet.

**Comment by DanielRuf** [comment]
See https://caniuse.com/css-focus-visible regarding a better approach. Or some polyfill or some solution like the `what-input` package.

**Comment by DanielRuf** [comment]
The "ugly" black border: that is Chrome's / Chromium's new focus ring design and their "fault".
We should never remove it but probably style it differently.

**Comment by rdevroede** [comment]
The issue is that the focus ring appears on mouse over or out (can't remember) and stays visible. It should never appear for desktop mouse usage, only on actual focus with the keyboard tab key. Even when someone would want it to appear on a mouse event, it should disappear on mouse out.

**Comment by JayDijkstra** [comment]
It seems that this issue only occurs when using the option `hover: true` on a M.dropdown() element. By disabling this, it seems to resolve this issue for me, however this is only suitable when you actually do not need the hover functionality on our dropdown buttons.  Maby this is some valuable information for you.
